rollup/sync_service.go

```diff
@@ -951,20 +1007,47 @@ func (s *SyncService) ProcessSequencerBatchAppendedLog(ctx context.Context, ethl
 					return fmt.Errorf("Unexpected type when parsing ctc tx eip155: %T", ctcTx.tx)
 				}
 				nonce, gasLimit := uint64(eip155.nonce), uint64(eip155.gasLimit)
-				to, l1TxOrigin := eip155.target, common.Address{}
+				to := eip155.target
 				gasPrice := new(big.Int).SetUint64(uint64(eip155.gasPrice))
 				data := eip155.data
-				tx = types.NewTransaction(nonce, to, big.NewInt(0), gasLimit, gasPrice, data, &l1TxOrigin, nil, types.QueueOriginSequencer, types.SighashEIP155)
+				l1BlockNumber := element.BlockNumber
+				// Set the L1TxOrigin to `nil`
+				tx = types.NewTransaction(nonce, to, big.NewInt(0), gasLimit, gasPrice, data, nil, l1BlockNumber, types.QueueOriginSequencer, types.SighashEIP155)
 				tx.SetIndex(index)
+				tx.SetL1Timestamp(element.Timestamp.Uint64())
 				// `WithSignature` accepts:
 				// r || s || v where v is normalized to 0 or 1
 				tx, err = tx.WithSignature(s.signer, eip155.Signature[:])
 				if err != nil {
 					return fmt.Errorf("Cannot add signature to eip155 tx: %w", err)
 				}
 				log.Debug("Deserialized CTC EIP155 transaction", "index", index, "to", tx.To().Hex(), "gasPrice", tx.GasPrice().Uint64(), "gasLimit", tx.Gas())
+			case CTCTransactionTypeEthSign:
+				// The signature is deserialized so the god key does not need to
+				// sign in this case.
+				ethsign, ok := ctcTx.tx.(*CTCTxEthSign)
+				if !ok {
+					return fmt.Errorf("Unexpected type when parsing ctc tx eip155: %T", ctcTx.tx)
+				}
+				nonce, gasLimit := uint64(ethsign.nonce), uint64(ethsign.gasLimit)
+				to := ethsign.target
+				gasPrice := new(big.Int).SetUint64(uint64(ethsign.gasPrice))
+				data := ethsign.data
+				l1BlockNumber := element.BlockNumber
+				// Set the L1TxOrigin to `nil`
+				tx = types.NewTransaction(nonce, to, big.NewInt(0), gasLimit, gasPrice, data, nil, l1BlockNumber, types.QueueOriginSequencer, types.SighashEthSign)
+				tx.SetIndex(index)
+				tx.SetL1Timestamp(element.Timestamp.Uint64())
+				// `WithSignature` accepts:
+				// r || s || v where v is normalized to 0 or 1
+				tx, err = tx.WithSignature(s.signer, ethsign.Signature[:])
+				if err != nil {
+					return fmt.Errorf("Cannot add signature to ethsign tx: %w", err)
+				}
+				log.Debug("Deserialized CTC EthSign transaction", "index", index, "to", tx.To().Hex(), "gasPrice", tx.GasPrice().Uint64(), "gasLimit", tx.Gas())
 			default:
-				// This should never happen
+				// This should never happen, the contract will never
+				// allow an unknown type to be in a valid transaction.
 				return fmt.Errorf("Unknown tx type: %x", element.TxData)
```

Comment:
We definitely need a ticket for allowing the verifier to sync unknown transactions with transaction types. We should decode this as approx:

```
to: sequencerEntrypoint
calldata: allDataSupplied
gasLimit: MAX_GAS
gasPrice: 0
... all other fields use default values
```

Reply:
Added a comment to address this